### ISSUE TITLE
How to add envoy prestop hook bash command?

### ISSUE THREAD

**Issue by vuducmanh11** [kubesphere/kubesphere#4479]
Microservice using isito get 503  when rolling update microservice. Follow [issue](https://github.com/istio/istio/issues/7136) therefore i need to add envoy prestop hook. I edited value of configmap follow ur *http://kubesphere_ui_url:clusters/default/projects/istio-system/configmaps/istio-sidecar-injector-1-6-10/detail*:
```
...........................
         "lifecycle": {
            "preStop": {
              "exec": {
                "command":  "[\"/bin/sh\", \"-c\", \"while [ $(netstat -plunt | grep tcp | grep -v envoy | wc -l | xargs) -ne 0 ]; do sleep 1; done\"]"
              }
            }
          },
...........................
```
After the change when I created microservice, I got below error log of pod istiod:
```

2021-11-26T11:54:23.888364Z     error   Injection data: err=failed parsing generated injected YAML (check Istio sidecar injector configuration): error unmarshaling JSON: json: cannot unmarshal string into Go struct field ExecAction.Contai
ners.lifecycle.preStop.exec.command of type []string spec=

containers:
- name: istio-proxy
  image: "istio/proxyv2:1.6.10"
  ports:
  - containerPort: 15090
    protocol: TCP
    name: http-envoy-prom
  args:
  - proxy
  - sidecar
  - --domain
  - $(POD_NAMESPACE).svc.cluster.local
  - --serviceCluster
  - "ratings.$(POD_NAMESPACE)"
  - --proxyLogLevel=warning
  - --proxyComponentLogLevel=misc:error
  - --trust-domain=cluster.local
  - --concurrency
  - "2"
  lifecycle:
    preStop:
      exec:
        command: '["/bin/sh", "-c", "while [ $(netstat -plunt | grep tcp | grep -v envoy
          | wc -l | xargs) -ne 0 ]; do sleep 1; done"]'

```

**Comment by zryfish** [comment]
Currently, pre-stop/post-start isn't supported on the KubeSphere console. I'll make it a feature request, see if there are contributors who are willing to do this.

/kind feature-request
/help

**Comment by vuducmanh11** [comment]
Now, i'm using kubectl command to edit configmap istio-sidecar-injector for adding prestophook of istio-proxy container
```
kubectl -n istio-system edit cm istio-sidecar-injector-1-6-10
...
  values: |-
     {
        "global": {
          "includeIPRanges": "*",
          "lifecycle": {
            "preStop": {
              "exec": {
                "command": [
                  "/bin/sh",
                  "-c",
                  "sleep 100"
                ]
              }
            }
          },
...
```

**Comment by XiaohuiSept** [comment]
Drawing design as follow:
https://www.figma.com/file/ZRvI25Lr3phV5a9mNu6jL9/KubeSphere-3.2.0?node-id=1118%3A373106

**Comment by wenxin-01** [comment]
issue fixed in `v3.3.0-alpha.2`

Hi @vuducmanh11, you can edit lifecycle in containers setting or create resources page.

![image](https://user-images.githubusercontent.com/68640256/167233872-dd59ffe2-e279-4dfa-a54a-f1e6be387820.png)

**Comment by adrianoaguiar** [comment]
> Drawing design as follow: https://www.figma.com/file/ZRvI25Lr3phV5a9mNu6jL9/KubeSphere-3.2.0?node-id=1118%3A373106

@XiaohuiSept the link https://www.figma.com/file/ZRvI25Lr3phV5a9mNu6jL9/KubeSphere-3.2.0?node-id=1118%3A373106 is not available to my user figma.

Do you can help me?
